src/useLocalStore.ts

```diff
@@ -11,9 +11,26 @@ function wrapInTransaction(fn: Function) {
     }
 }
 
-export function useLocalStore<T>(initializer: () => T): T {
+export function useLocalStore<T>(initializer: (props?: any) => T, props?: any): T {
+    const [res] = useState(() => (props ? observable(props, {}, { deep: false }) : {}))
+
+    if (typeof props !== "undefined") {
```

Comment:
Let's reuse `useAsObservableSource`

Reply:
As mentioned in https://github.com/mobxjs/mobx-react-lite/pull/142#issuecomment-488320376 it can't really be done

Reply:
I think the common stuff can still be extracted, just parameterize the hook name in the error message, and make sure the call from this hook can deal with the absence of the thing?

Reply:
This seems like overkill to avoid copying 2 lines of code (rest is just development-only). Parametrizing messages will also make typescript type of `useAsObservableSource` very ugly and I'll need to check whether `current` is defined each time before I do `Object.assign` which currently is not necessary as `useAsObservableSource` in non-development mode just assumes that current is an object.

Reply:
I'll commit in short while code that reduces number of hook calls inside `useLocalStore` to 1 and still supports passing custom props dependencies